### PR TITLE
doc: add guide on enabling Travis under nodejs

### PR DESCRIPTION
**Pull request by joyeecheung**
@nodejs/automation @nodejs/github-bot @nodejs/build 

This documents how to enable Travis in a repository under the Node.js Github organization.
I'll open a separate PR in the TSC repo pointing to this guide after this one lands.

### CODE REVIEW

**Review by Tiriel — APPROVED** [review]

File: enable-travis-under-nodejs.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# How to enable Travis in a repository under the Node.js Github organization
+
+Since we don't allow third-party access to the Node.js Github organization
+except our own bots, when a repository is transferred into the Node.js
+foundation, Travis won't be able to update the build status of Pull requests
+automatically. Instead, we need to use our own github bot to pull the status
+from Travis and send the update it Github. This is what needs to be done:
```
Comment (vsemozhetbyt) [inline enable-travis-under-nodejs.md]: it Github -> in Github?

File: enable-travis-under-nodejs.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# How to enable Travis in a repository under the Node.js Github organization
```
Comment (vsemozhetbyt) [inline enable-travis-under-nodejs.md]: Github / github -> GitHub (all over the text)?

File: enable-travis-under-nodejs.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+### Step 1: Travis setup
+
+Make sure Travis can listen to the push and pull request events of the
+repository. Go to "Settings -> Integrations & services" page of the
+repository. If it has previously configured Travis, then Travis should be
+listed under "Services". If not, click on the dropdown "Add service", and
+search for Travis to add it.
+
+The service does not really have to be configured to enable travis. All of the
```
Comment (vsemozhetbyt) [inline enable-travis-under-nodejs.md]: travis -> Travis ?

File: enable-travis-under-nodejs.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+* Finally, check "Active" and save the webhook. Github would send a test
+  payload to the bot, if it came back green, the bot is set up with the
+  repository.
+
+### Step 4: update the bot
+
+Open a pull request to https://github.com/nodejs/github-bot , adding the
+repository to the `enabledRepos` in
+https://github.com/nodejs/github-bot/blob/master/scripts/display-travis-status.js,
+so the bot would not ignore the events from your repository ans start sending
```
Comment (vsemozhetbyt) [inline enable-travis-under-nodejs.md]: ans -> and ?

File: enable-travis-under-nodejs.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+The bot works like this:
+
+1. when a pull request has been opened or updated,
+Github will send an event to both Travis (configured in step 1) and the bot
+(step 3).
+2. Travis would start the build, and the bot would start polling Travis
+for build status (this requires step 4 otherwise the bot would ignore the
+event).
+3. When the bot find the build matching the last commit in the PR, it
```
Comment (vsemozhetbyt) [inline enable-travis-under-nodejs.md]: find -> finds

File: enable-travis-under-nodejs.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
+# How to enable Travis in a repository under the Node.js Github organization
+
+Since we don't allow third-party access to the Node.js Github organization
+except our own bots, when a repository is transferred into the Node.js
+foundation, Travis won't be able to update the build status of Pull requests
```
Comment (vsemozhetbyt) [inline enable-travis-under-nodejs.md]: A nit: unify upper / lower case in "Pull request[s]" / "pull request[s]"

File: enable-travis-under-nodejs.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+automatically. Instead, we need to use our own github bot to pull the status
+from Travis and send the update it Github. This is what needs to be done:
+
+### Step 1: Travis setup
+
+Make sure Travis can listen to the push and pull request events of the
+repository. Go to "Settings -> Integrations & services" page of the
+repository. If it has previously configured Travis, then Travis should be
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: If it has -> If it has been

File: enable-travis-under-nodejs.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+listed under "Services". If not, click on the dropdown "Add service", and
+search for Travis to add it.
+
+The service does not really have to be configured to enable travis. All of the
+following configurations works:
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: configurations works -> configurations work

File: enable-travis-under-nodejs.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+On the "Settings -> Webhooks" page, add a new webhook for the bot:
+
+* Payload URL: `http://github-bot.nodejs.org:3333/hooks/github`
+* Content type: `application/json`
+* Secret: can be obtained in the https://github.com/nodejs/secrets repository,
+  or ask someone from the build working group to add the webhook for you.
+* In "Which events would you like to trigger this webhook?", select "Let me
+  select individual events.", check "Push" and "Pull Request".
+* Finally, check "Active" and save the webhook. Github would send a test
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: Github would -> GitHub will

File: enable-travis-under-nodejs.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+  payload to the bot, if it came back green, the bot is set up with the
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: payload to the bot, if it came back green, -> payload to the bot. If it comes back green,

File: enable-travis-under-nodejs.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+  repository.
+
+### Step 4: update the bot
+
+Open a pull request to https://github.com/nodejs/github-bot , adding the
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: Extra space before `,`

File: enable-travis-under-nodejs.md (hunk 4 above)
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: would not -> will not

File: enable-travis-under-nodejs.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+updates.
+
+### How it works
+
+The bot works like this:
+
+1. when a pull request has been opened or updated,
+Github will send an event to both Travis (configured in step 1) and the bot
+(step 3).
+2. Travis would start the build, and the bot would start polling Travis
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: would start -> will start

File: enable-travis-under-nodejs.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+for build status (this requires step 4 otherwise the bot would ignore the
+event).
+3. When the bot find the build matching the last commit in the PR, it
+would update the build status of that PR as "pending" (this update requires
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: would update -> will update

File: enable-travis-under-nodejs.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+step 2).
+4. When a poll came back as a successful build, the bot would update the
```
Comment (riyadhalnur) [inline enable-travis-under-nodejs.md]: poll came back -> poll comes back
the bot would -> the bot will

**Review by riyadhalnur — CHANGES_REQUESTED** [review]

**Review by mhdawson — APPROVED** [review]
LGTM with suggested changes by @vsemozhetbyt

File: enable-travis-under-nodejs.md (hunk 7 above)
Comment (joyeecheung) [reply enable-travis-under-nodejs.md]: `If it has been previously configured Travis` sounds a bit weird as well. Maybe `If it has previously been integrated with Travis`?

**Comment by maclover7** [comment]
@joyeecheung Should this live in `nodejs/build`? I feel like this has to do with CI / provisioning resources, and would maybe be a better fit there?

File: enable-travis-under-nodejs.md (hunk 7 above)
Comment (joyeecheung) [reply enable-travis-under-nodejs.md]: Hmm...or `If Travis has been integrated with the repository before`

**Comment by joyeecheung** [comment]
@maclover7 It's not related to our own CI resources though, except the bot part, and the bot is managed by automation team now, so I guess it makes sense to put it here (this was suggested by @mhdawson ).

**Comment by joyeecheung** [comment]
^ Although I am not quite sure about "the bot is managed by automation team" part either...so...

As long as there is a document about this I am fine with it being anywhere.

**Comment by joyeecheung** [comment]
@mhdawson @vsemozhetbyt @riyadhalnur @Tiriel I've updated the guide to address the comments, PTAL, thanks

**Review by riyadhalnur — APPROVED** [review]
LGTM 👍

**Review by vsemozhetbyt — APPROVED** [review]

**Pull request merged**